tap_intercom/intercom.py

```diff
@@ -0,0 +1,94 @@
+import requests
+from ratelimit import limits
+import ratelimit
+import singer
+import backoff
+import datetime
+from typing import Dict
+
+LOGGER = singer.get_logger()
+ONE_MINITE = 60
+
+
+class Intercom:
+    BASE_URL = "https://api.intercom.io"
+    DATA_FIELD = {
+        "companies": "data",
+        "contacts": "data",
+        "tags": "data",
+        "conversations": "conversations",
+        "segments": "segments",
+    }
+
+    def __init__(self, access_token, tap_stream_id):
+        self.SESSION = requests.Session()
+        self.access_token = access_token
+        self.tap_stream_id = tap_stream_id
+
+    def get_records(self):
+        per_page = 150 if self.tap_stream_id == "contacts" else 60
+        replication_path = ["updated_at"]
+        params = {"per_page": per_page}
+
+        url = f"{self.BASE_URL}/{self.tap_stream_id}"
+        for record in self.paginate(url, params=params):
+
+            replication_value = self.unixseconds_to_datetime(
+                self.get_value(record, replication_path)
+            )
+
+            yield record, replication_value
+
+    def get_value(self, obj: dict, path_to_replication_key=None, default=None):
```

Comment:
is path to `path_to_replication_key` really optional? Won't it break if this function returns `None`?

Reply:
if replication value is none. the tap will still write this record. This is for those record that need to be synced FULL_TABLE.

Reply:
Makes sense!